gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -197,6 +201,22 @@ protected void shutdownMetrics() throws IOException {
    */
   protected void createMetrics() {
     this.messagesRead = this.metricContext.counter(RuntimeMetrics.GOBBLIN_KAFKA_HIGH_LEVEL_CONSUMER_MESSAGES_READ);
+    this.queueSizeGauges = new ContextAwareGauge[numThreads];
+    this.queueSizeGaugeValues = new AtomicIntegerArray(numThreads);
+    for (int i=0; i < numThreads; i++) {
+      int finalI = i;
```

Comment:
I see what you're doing... but it may not immediately register to everyone... especially since `finalI` is not `final int`.  please add a comment calling attention to variable capture by the closure